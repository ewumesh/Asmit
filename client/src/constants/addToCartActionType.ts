 export enum AddToCartActionType {
    CLEAR_ALL_DATA_CART= 'CLEAR_ALL_DATA_CART',
    PRODUCT_TO_CART = 'PRODUCT_TO_CART',
    ORDER_TO_CART = 'ORDER_TO_CART',
    UPDATE_PRODUCT_TO_CART = 'UPDATE_PRODUCT_TO_CART',
    REMOVE_PRODUCT_FROM_CART = 'REMOVE_PRODUCT_FROM_CART'
}
