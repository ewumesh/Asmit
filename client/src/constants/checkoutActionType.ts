 export enum CheckoutActionType {
    CLEAR_ALL_DATA_CHECKOUT= 'CLEAR_ALL_DATA_CHECKOUT',
    ACTIVE_STEP = 'ACTIVE_STEP'
}
