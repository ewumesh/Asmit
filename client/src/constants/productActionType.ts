 export enum ProductActionType {

    ADD_IMAGE_PRODUCT = 'ADD_IMAGE_PRODUCT',
    HAS_MORE_DATA_STREAM = 'HAS_MORE_DATA_STREAM',
    NOT_MORE_DATA_STREAM = 'NOT_MORE_DATA_STREAM',
    REQUEST_PAGE_STREAM = 'REQUEST_PAGE_STREAM',
    LAST_PRODUCT_STREAM = 'LAST_PRODUCT_STREAM',
    HAS_MORE_DATA_PROFILE = 'HAS_MORE_DATA_PROFILE',
    NOT_MORE_DATA_PROFILE = 'NOT_MORE_DATA_PROFILE',
    REQUEST_PAGE_PROFILE = 'REQUEST_PAGE_PROFILE',
    LAST_PRODUCT_PROFILE = 'LAST_PRODUCT_PROFILE',
    ADD_VIDEO_PRODUCT = 'ADD_VIDEO_PRODUCT',
    ADD_PRODUCT_DETAIL = 'ADD_PRODUCT_DETAIL',
    UPDATE_PRODUCT = 'UPDATE_PRODUCT',
    UPDATE_PRODUCT_COMMENTS = 'UPDATE_PRODUCT_COMMENTS',
    UPDATE_PRODUCT_VOTES = 'UPDATE_PRODUCT_VOTES',
    DELETE_PRODUCT = 'DELETE_PRODUCT',
    ADD_LIST_PRODUCT = 'ADD_LIST_PRODUCT',
    ADD_LIST_PRODUCT_SEARCH_TEXT = 'ADD_LIST_PRODUCT_SEARCH_TEXT',
    ADD_LIST_PRODUCT_CATEGORY = 'ADD_LIST_PRODUCT_CATEGORY',
    ADD_LIST_PRODUCT_ATTRIBUTES = 'ADD_LIST_PRODUCT_ATTRIBUTES',
    CLEAR_ALL_DATA_PRODUCT = 'CLEAR_ALL_DATA_PRODUCT',
    CLEAR_SEARCH_DATA_PRODUCT = 'CLEAR_SEARCH_DATA_PRODUCT',
    CLEAR_CATEGORY_DATA_PRODUCT = 'CLEAR_CATEGORY_DATA_PRODUCT',
    PRODUCTS_PAGE_COUNT = 'PRODUCTS_PAGE_COUNT'
}
