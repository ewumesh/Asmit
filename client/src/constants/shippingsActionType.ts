 export enum ShippingsActionType {

    ADD_IMAGE_PRODUCT = 'ADD_IMAGE_PRODUCT',
    HAS_MORE_DATA_STREAM = 'HAS_MORE_DATA_STREAM',
    NOT_MORE_DATA_STREAM = 'NOT_MORE_DATA_STREAM',
    REQUEST_PAGE_STREAM = 'REQUEST_PAGE_STREAM',
    LAST_PRODUCT_STREAM = 'LAST_PRODUCT_STREAM',
    HAS_MORE_DATA_PROFILE = 'HAS_MORE_DATA_PROFILE',
    NOT_MORE_DATA_PROFILE = 'NOT_MORE_DATA_PROFILE',
    REQUEST_PAGE_PROFILE = 'REQUEST_PAGE_PROFILE',
    LAST_PRODUCT_PROFILE = 'LAST_PRODUCT_PROFILE',
    ADD_VIDEO_PRODUCT = 'ADD_VIDEO_PRODUCT',
    ADD_PRODUCT = 'ADD_PRODUCT',
    UPDATE_PRODUCT = 'UPDATE_PRODUCT',
    UPDATE_PRODUCT_COMMENTS = 'UPDATE_PRODUCT_COMMENTS',
    UPDATE_PRODUCT_VOTES = 'UPDATE_PRODUCT_VOTES',
    DELETE_PRODUCT = 'DELETE_PRODUCT',
    ADD_LIST_SHIPPING_RIGIONS = 'ADD_LIST_SHIPPING_RIGIONS',
    ADD_LIST_SHIPPING_RATES = 'ADD_LIST_SHIPPING_RATES',
    CLEAR_ALL_DATA_PRODUCT = 'CLEAR_ALL_DATA_PRODUCT',
    ADD_SHIPPING_ADDRESS = 'ADD_SHIPPING_ADDRESS'
}
