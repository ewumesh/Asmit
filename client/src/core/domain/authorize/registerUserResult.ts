import { BaseDomain } from 'core/domain/common'

export class RegisterUserResult extends BaseDomain {

  private _uid: string
  private _name: string
  private _email: string
  constructor (uid: string, name: string, email: string) {
    super()

    this._uid = uid
    this._name = name
    this._email = email
  }
    /**
     * User identifier
     *
     * @type {string}
     * @memberof LoginUser
     */

  public get uid (): string {
    return this._uid
  }

  public get name (): string {
    return this._name
  }

  public get email (): string {
    return this._email
  }
}
