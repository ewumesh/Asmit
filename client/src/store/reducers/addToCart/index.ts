import { addToCartReducer } from './addToCartReducer'
export {
    addToCartReducer
}
