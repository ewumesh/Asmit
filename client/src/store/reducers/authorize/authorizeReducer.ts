// - Import react components
import { Map } from 'immutable'

import { AuthorizeActionType } from 'constants/authorizeActionType'

import { AuthorizeState } from './AuthorizeState'
import { IAuthorizeAction } from './IAuthorizeAction'

/**
 *  Authorize reducer
 * @param {object} state
 * @param {object} action
 */
export let authorizeReducer = (state = Map(new AuthorizeState() as any), action: IAuthorizeAction) => {
  const { payload } = action
  switch (action.type) {
    case AuthorizeActionType.LOGIN:
    return state
        .set('uid', payload.uid)
        .set('authed', payload.authed)
        .set('guest', false)
        .set('isVerifide', true)
        .set('displayName', payload.displayName)
        .set('email', payload.email)

    case AuthorizeActionType.LOGOUT:
      return state
        .set('uid', 0)
        .set('authed', false)
        .set('guest', true)
        .set('isVerifide', false)
        .set('logoutMessage', payload.msg)
        .set('logoutStatus', payload.status)
        .set('displayName', '')
        .set('email', '')

    case AuthorizeActionType.SIGNUP:
      return state
      .set('uid', payload.userId)
      .set('displayName', payload.fullName)
      .set('email', payload.email)
    case AuthorizeActionType.UPDATE_PASSWORD:
      return state
      .set('updatePassword', payload.updatePassword)
    default:
      return state

  }

}
