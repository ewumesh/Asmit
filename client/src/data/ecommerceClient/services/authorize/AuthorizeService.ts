import 'reflect-metadata'

import { injectable } from 'inversify'
import axios from 'axios'

import { IAuthorizeService } from 'core/services/authorize'
import { LoginUser, RegisterUserResult } from 'core/domain/authorize'
import { OAuthType } from 'core/domain/authorize/oauthType'
import { SocialError } from 'core/domain/common'
import { User } from 'core/domain/users'
import { getLoginURL, getSignupURL, getLogoutURL, getUserInfoURL } from 'data/ecommerceClient'

/**
 * Firbase authorize service
 *
 * @export
 * @class AuthorizeService
 * @implements {IAuthorizeService}
 */
@injectable()
export class AuthorizeService implements IAuthorizeService {

  /**
   * Login the user
   */
  public login = (email: string, password: string) => {
    return new Promise<LoginUser>((resolve, reject) => {
      const data = {email: email,password: password}
      const config = { headers: {'Content-Type': 'application/json'}, withCredentials: true }
      axios.post(getLoginURL, data, config)
      .then((result) => {
        // The signed-in user info.
        resolve(new LoginUser(result.data.customerId, true, '0' , result.data.name, result.data.email))
      }).catch(error => {
        if (error.message.includes('401')) {
          reject(new SocialError('401', 'Please check your email and password again'))
        } else {
          reject(new SocialError(error.code, error.message))
        }

      })

    })
  }

  /**
   * Logs out the user
   *
   * @returns {Promise<void>}
   * @memberof IAuthorizeService
   */
  public logout: () => Promise<{msg: string, status: boolean}> = () => {
    return new Promise<{msg: string, status: boolean}>((resolve, reject) => {
      const data = {}
      const config = { headers: {'Content-Type': 'application/json'}, withCredentials: true }
      axios.post(getLogoutURL, data, config)
      .then((result) => {
        resolve(result.data)

      }).catch(error => {
        reject(new SocialError(error.code, error.message))
      })
    })
  }

  /**
   * Register a user
   */
  public registerUser = (registerUser: User) => {
    return new Promise<RegisterUserResult>((resolve, reject) => {
      const data = {email: registerUser.email as string,password: registerUser.password as string, name: registerUser.fullName as string}
      const config = { headers: {'Content-Type': 'application/json'},  withCredentials: true }

      axios.post(getSignupURL, data, config)
      .then((result) => {
        // The signed-in user info.
         resolve(new RegisterUserResult(result.data.customerId, result.data.name, result.data.email))

      }).catch(error => {
        reject(new SocialError(error.code, error.message))
      })

    })
  }

  /**
   * Update user password
   *
   * @returns {Promise<void>}
   * @memberof IAuthorizeService
   */
  public updatePassword: (newPassword: string) => Promise<void> = (newPassword) => {

    return new Promise<void>((resolve, reject) => {
        // TODO
        resolve()
    })
  }

  /**
   * On user authorization changed event
   *
   * @memberof IAuthorizeService
   */
  public onAuthStateChanged: (callBack: (isVerifide: boolean, user: User) => void) => any = (callBack) => {
      // TODO - Need to get real user
      const config = { headers: {'Content-Type': 'application/json'}, withCredentials: true }
      axios.get(getUserInfoURL, config)
      .then((result) => {
        // The signed-in user info.
        let user = new User()
        user.email = result.data.email
        user.userId = result.data.customerId
        user.fullName = result.data.name
        callBack(true, user)
      }).catch(error => {
        callBack(false, new User())
      })

  }

  /**
   * Reset user password
   *
   * @memberof AuthorizeService
   */
  public resetPassword: (email: string) => Promise<void> = (email) => {
    return new Promise<void>((resolve, reject) => {
        // TODO
        resolve()
    })
  }

  /**
   * Send verfication email to user email
   *
   * @memberof AuthorizeService
   */
  public sendEmailVerification: () => Promise<void> = () => {
    return new Promise<void>((resolve, reject) => {
          // TODO
          resolve()
    })
  }

  public loginWithOAuth: (type: OAuthType) => Promise<LoginUser> = (type) => {
    return new Promise<LoginUser>((resolve, reject) => {
        // TODO
      resolve()
    })
  }

  /**
   * Store user information
   *
   * @private
   * @memberof AuthorizeService
   */
  private storeUserInformation = (userId: string, email: string, fullName: string, avatar: string) => {
    return new Promise<RegisterUserResult>((resolve, reject) => {
          // TODO
          resolve(new RegisterUserResult(userId, fullName, email))
    })
  }

  /**
   * Store user provider information
   *
   * @private
   * @memberof AuthorizeService
   */
  private storeUserProviderData = (
    userId: string,
    email: string,
    fullName: string,
    avatar: string,
    providerId: string,
    accessToken: string
  ) => {
    return new Promise<RegisterUserResult>((resolve, reject) => {
        // TODO
        resolve(new RegisterUserResult(userId, fullName, email))
    })
  }
}
