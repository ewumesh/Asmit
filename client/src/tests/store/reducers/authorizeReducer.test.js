import { authorizeReducer } from '../../../store/reducers/authorize'
import { AuthorizeState } from '../../../store/reducers/authorize/AuthorizeState'
import { AuthorizeActionType } from '../../../constants/authorizeActionType'
import { Map } from 'immutable'

describe('authorize reducer', () => {
  it('should return the initial state', () => {
    expect(authorizeReducer(undefined, {})).toMatchSnapshot()
  })

  it('should handle LOGIN', () => {
    let state = Map(new AuthorizeState())
    expect(
      authorizeReducer(state, {
        type: AuthorizeActionType.LOGIN,
        payload: { authed: true, isVerifide: true, uid: '1' }
      })
    ).toMatchSnapshot()
  })

  it('should handle SIGNUP', () => {
    let state = Map(new AuthorizeState())
    expect(
      authorizeReducer(state, {
        type: AuthorizeActionType.SIGNUP,
        payload: { userId: '1' }
      })
    ).toMatchSnapshot()
  })

  it('should handle Logout', () => {
    let state = Map(new AuthorizeState())
    expect(
      authorizeReducer(state, {
        type: AuthorizeActionType.LOGOUT,
        payload: { msg: 'User logout successfully1', status: true }
      })
    ).toMatchSnapshot()
  })
})
