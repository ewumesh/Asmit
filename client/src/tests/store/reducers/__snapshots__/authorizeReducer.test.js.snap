// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authorize reducer should handle LOGIN 1`] = `
Immutable.Map {
  uid: "1",
  authed: true,
  isVerifide: true,
  updatePassword: false,
  guest: false,
  displayName: undefined,
  email: undefined,
}
`;

exports[`authorize reducer should handle Logout 1`] = `
Immutable.Map {
  uid: 0,
  authed: false,
  isVerifide: false,
  updatePassword: false,
  guest: true,
  logoutMessage: "User logout successfully1",
  logoutStatus: true,
}
`;

exports[`authorize reducer should handle SIGNUP 1`] = `
Immutable.Map {
  uid: "1",
  authed: false,
  isVerifide: false,
  updatePassword: false,
  guest: false,
  displayName: undefined,
  email: undefined,
}
`;

exports[`authorize reducer should return the initial state 1`] = `
Immutable.Map {
  uid: 0,
  authed: false,
  isVerifide: false,
  updatePassword: false,
  guest: false,
}
`;
